docs/en/observability/profiling-upgrade.asciidoc

```diff
@@ -54,10 +56,359 @@ Integrations Server. To do this:
+            "max_age": "30d",
+            "min_docs": 1
+          },
+          "set_priority": {
+            "priority": 100
+          },
+          "readonly": {}
+        }
+      },
+      "warm": {
+        "min_age": "30d",
+        "actions": {
+          "set_priority": {
+            "priority": 50
+          },
+          "forcemerge": {
+            "max_num_segments": 1
+          }
+        }
+      },
+      "delete": {
+        "min_age": "60d",
+        "actions": {
+          "delete": {}
+        }
+      }
+    },
+    "_meta": {
+      "description": "default policy for Elastic Universal Profiling",
+      "managed": true,
+      "version": 1
+    }
+  }
+}'
+
+  echo "Renaming indices to new naming scheme..."
+  rename_kv_index "profiling-stackframes"
+  rename_kv_index "profiling-stacktraces"
+  rename_kv_index "profiling-executables"
+
+  # profiling-symbols-private may or may not be present
+  status_code=$(curl -s -o /dev/null -w "%{http_code}" -I -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/profiling-symbols-private")
+  if [ "$status_code" == "200" ]; then
+    rename_regular_index "profiling-symbols-private"
+  fi
+
+  echo "Renaming existing fields..."
+  curl -s -f -o /dev/null -X PUT -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/profiling-events-*/_mapping" -H 'Content-Type: application/json' -d'
+  {
+    "properties": {
+      "profiling.project.id": {
+        "type": "alias",
+        "path": "service.name"
+      }
+    }
+  }'
+
+  echo "Enabling Universal Profiling index template management and disabling wildcard deletes..."
+  curl -s -f -o /dev/null -X PUT -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/_cluster/settings" -H 'Content-Type: application/json' -d'
+  {
+    "persistent": {
+      "xpack.profiling.templates.enabled": true,
+      "action.destructive_requires_name": true
+    }
+  }'
+
+  echo "Reenabling ILM..."
+  curl -s -f -o /dev/null -X POST -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/_ilm/start"
+
+  printf "Checking cluster health ... "
+  local status=$(curl -s -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/_cluster/health?wait_for_status=green&wait_for_no_initializing_shards=true&wait_for_no_relocating_shards=true" | jq -r ".status")
+  if [ "$status" == "red" ]; then
+    printf "[ERROR: status is red]\n"
+  else
+    printf "[OK: status is %s]\n" "$status"
+  fi
+
+  {
+    echo "Indices and aliases after migration"
+    curl -s -f -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/_cat/indices/*profiling*?v&s=index&expand_wildcards=all"
+    curl -s -f -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/_cat/aliases/*profiling*?v&expand_wildcards=all"
+  } >> "elastic_universal_profiling_migration_log.txt"
+}
+
+function main {
+  check_preconditions
+  ask_user_prerequisites
+  sanity_check_cluster
+  migrate
+
+  echo "Migration has finished. Please perform the following tasks:"
+  echo ""
+  echo "* Start profiling data ingestion"
+  echo "* Enable Elastic Universal Profiling in Kibana"

```

Comment:
Do you think we can remove this line? Profiling is enabled by default in all Cloud deployments since 8.9.

Reply:
Good catch. I've reworded the entire statement now to:

> Migration has finished. You can now start profiling data ingestion.